### ISSUE TITLE
Unnecessary parentheses - false positive

### ISSUE THREAD

**Issue by SpacePossum** [kalessil/phpinspectionsea#579]
Hi!

I did a search through the `issues` and `PR's` but I think there is a small issue with `Unnecessary parentheses` detection.
I'm using
- Php Inspections (EA extended) 2.3.10 
- PHPStorm 2016.3.3
- PHP 7.0.23-1+ubuntu14.04.1+deb.sury.org+1

The case:
```php
function A()
{
    // ...
    return ('My\\Project\\'.$classMap[$shortName])($arg1);
}
```
